src/cosalib/qemuvariants.py

```diff
@@ -312,18 +312,21 @@ def mutate_image(self):
         elif self.compression is not None:
             sha256 = sha256sum_file(final_img)
             size = os.stat(final_img).st_size
-            temp_path = f"{final_img}.tmp"
 
+            # the compressor embeds the input filename in its output, so
+            # strip the .gz or .zip suffix from the uncompressed file
```

Comment:
Is there a page we can link to that describes the behavior where the compressor embeds the input filename in its output and that this matters for zip?

Should we put more of the commit message (good context) here?

Reply:
Updated the comment.  The zip behavior is because it's a container format like tar.